src/jit/lowerxarch.cpp

```diff
@@ -3093,6 +3097,41 @@ void Lowering::SetIndirAddrOpCounts(GenTreePtr indirTree)
     }
 }
 
+//------------------------------------------------------------------------
+// TreeNodeInfoInitSelect: Set the NodeInfo for a GT_SELCC
+//
+// Arguments:
+//    select - the select node
+//
+
+void Lowering::TreeNodeInfoInitSelect(GenTreeSelectCC* select)
+{
+    TreeNodeInfo* info = &(select->gtLsraInfo);
+
+    info->srcCount = 2;
+    info->dstCount = 1;
+
+    GenTree* op2 = select->gtGetOp2();
+
+    if (op2->isMemoryOp())
+    {
+        //
+        // There's no byte-sized CMOV so we can't contain a memory operand unless it's
+        // TYP_INT (and TYP_LONG on x64). There's a word-sized CMOV but to use it we
+        // we would likely need to widen the result of CMOV the same way a GT_IND does.
+        //
+
+        if (varTypeIsIntOrI(op2))
+        {
+            MakeSrcContained(select, op2);
+        }
+    }
+    else
+    {
+        SetRegOptional(op2);
```

Comment:
I may be trying to hard here. The compares generated by `LowerCompare` have type TYP_BYTE and there's no byte sized CMOV. If the compare tree is spilled (and I've seen that happening) then the second operand of CMOV will be a temp and CMOV will read 4 bytes out of it. As far as I can tell this works fine because spill temps are 4 or 8 bytes. Concerns?